### ISSUE TITLE
本地使用vllm启动，效果跟官网的demo不一样

### ISSUE THREAD

**Issue by design365** [opendatalab/MinerU#3682]
### 🔎 Search before asking | 提交之前请先搜索

- [x] I have searched the MinerU [Readme](https://github.com/opendatalab/MinerU) and found no similar bug report.
- [x] I have searched the MinerU [Issues](https://github.com/opendatalab/MinerU/issues) and found no similar bug report.
- [x] I have searched the MinerU [Discussions](https://github.com/opendatalab/MinerU/discussions) and found no similar bug report.

### 🤖 Consult the online AI assistant for assistance | 在线 AI 助手咨询

- [x] I have consulted the [online AI assistant](https://deepwiki.com/opendatalab/MinerU) but was unable to obtain a solution to the issue.

### Description of the bug | 错误描述

使用了“mineru-vllm-server --port 30000 ”启动后，
客户端使用“mineru -p  huaxue.pdf -o ./output/ -l ch -b vlm-http-client -u http://127.0.0.1:30000” 调用
pdf文件是几页图片，在你们官网demo是可以解析出来，但是使用上面这个命令解析的是空的。
请问为什么？

### How to reproduce the bug | 如何复现

本地与官网demo效果有出入

### Operating System Version| 操作系统版本

昇腾服务器910B3

### Python version | Python 版本

3.11

### Software version | 软件版本 (mineru --version)

2.5.4

### Device mode | 设备模式

npu

**Comment by myhloli** [comment]
pdf文件上传一下看看呢

**Comment by design365** [comment]
[huaxue.pdf](https://github.com/user-attachments/files/22844558/huaxue.pdf)

**Comment by design365** [comment]
> 你本地用昇腾 910B3 启动 vllm 后解析多页图片 PDF 得到空结果，主要原因很可能是 MinerU vllm 后端最近更新后，部分关键功能（比如自定义 logits processor）只在有 CUDA 的 NVIDIA GPU 上启用。如果没有 CUDA（比如昇腾/NPU），这些特性会被禁用，导致解析效果和官网 demo（通常用 NVIDIA GPU）不同，甚至输出为空。[相关代码逻辑和 PR 说明](https://github.com/opendatalab/MinerU/pull/3545)
> 
> 此外，MinerU 2.5.x 版本只支持 MinerU2.5 模型，且 vllm 后端和模型结构近期有较大调整。如果本地环境、依赖（如 mineru-vl-utils >=0.1.7）、模型路径或配置有误，也可能导致解析失败或输出为空。[版本和模型兼容性说明](https://github.com/opendatalab/MinerU/pull/3509)
> 
> 建议：
> 
> 1. 检查本地 MinerU、vllm 及 mineru-vl-utils 版本，确保为最新且与官网一致。
> 2. 确认模型已正确下载并加载，且配置参数（如 logits processor）未被错误禁用。
> 3. 查看 vllm server 和 CLI 的详细日志，排查是否有模型初始化、图片提取或 OCR 相关报错。
> 4. 如果硬件不兼容 vllm 后端，可以尝试用 pipeline 后端（多模型流水线，支持 CPU/NPU），效果更稳定。
> 
> 如需进一步定位，建议补充本地 MinerU 版本、Python 版本和详细日志。
> 
> _To reply, just mention [@dosu](https://go.dosu.dev/dosubot)._
> 
> How did I do? [Good](https://app.dosu.dev/response-feedback/3549200b-cfe7-4fc6-bc22-dddfd12ec111?feedback_type=great_response) | [Irrelevant](https://app.dosu.dev/response-feedback/3549200b-cfe7-4fc6-bc22-dddfd12ec111?feedback_type=irrelevant_answer) | [Incorrect](https://app.dosu.dev/response-feedback/3549200b-cfe7-4fc6-bc22-dddfd12ec111?feedback_type=incorrect_sources) | [Verbose](https://app.dosu.dev/response-feedback/3549200b-cfe7-4fc6-bc22-dddfd12ec111?feedback_type=too_verbose) | [Hallucination](https://app.dosu.dev/response-feedback/3549200b-cfe7-4fc6-bc22-dddfd12ec111?feedback_type=hallucination) | [Report 🐛](https://app.dosu.dev/response-feedback/3549200b-cfe7-4fc6-bc22-dddfd12ec111?feedback_type=bug_report) | [Other](https://app.dosu.dev/response-feedback/3549200b-cfe7-4fc6-bc22-dddfd12ec111?feedback_type=other)  [![Chat with Dosu](https://camo.githubusercontent.com/d9ffebb979729c9ef0fc4e3090dbae5b17cabbfec8afe378ada6b81913d6ad8a/68747470733a2f2f646f73752e6465762f646f73752d636861742d62616467652e737667)](https://app.dosu.dev/32d7cdb1-783f-42b5-8007-7cac8945fe20/ask?utm_source=github) [![Join Discord](https://camo.githubusercontent.com/d562c2df5e4567ea41d37f1d91a4da0839d8341037045b103b2da268998de17f/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f6a6f696e2d3538363546323f6c6f676f3d646973636f7264266c6f676f436f6c6f723d7768697465266c6162656c3d)](https://go.dosu.dev/discord-bot) [![Share on X](https://camo.githubusercontent.com/4649f705e2396d6f328dbc8e800a0509c9c17b3b20253b36706c393df124b93d/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f582d73686172652d626c61636b)](https://twitter.com/intent/tweet?text=%40dosu_ai%20helped%20me%20solve%20this%20issue!&url=https%3A//github.com/opendatalab/MinerU/issues/3682)

已经补充了环境信息

**Comment by design365** [comment]
<img width="553" height="64" alt="Image" src="https://github.com/user-attachments/assets/a1f9b21b-00aa-40c2-b1f1-4303c9048777" />

已经确认模型已正确下载并加载，请问配置参数（如 logits processor）如何启用？

**Comment by myhloli** [comment]
我们在n卡上测试是正常的，目前自定义 logits processor只在符合条件的n卡上启用，暂时还没有在npu上通过测试。

**Comment by design365** [comment]
> 我们在n卡上测试是正常的，目前自定义 logits processor只在符合条件的n卡上启用，暂时还没有在npu上通过测试。

您看我的启动和调用有什么不同呢？
启动命令： mineru-vllm-server --port 30000 
调用： mineru -p  huaxue.pdf -o ./output/ -l ch -b vlm-http-client -u http://127.0.0.1:30000
@myhloli

**Comment by myhloli** [comment]
目前推测可能是在npu上没启用自定义logits processor会有一些结果差异，但是也不排除可能是其他问题。

**Comment by design365** [comment]
> 目前推测可能是在npu上没启用自定义logits processor会有一些结果差异，但是也不排除可能是其他问题。

您是跟我这个一样的使用方式吗?

**Comment by myhloli** [comment]
使用方式一样，npu效果可能会和n卡有差异。
